### PR TITLE
[Flow EVM] improvements to COA proof verification

### PR DESCRIPTION
**Pull request by ramtinms**
This PR improves the COA proof verification function by returning a boolean from the invoke call, and handles errors properly (panic on fatal or backend errors)

### CODE REVIEW

File: fvm/evm/stdlib/contract.cdc [hunk 1]
```diff
@@ -233,6 +233,21 @@ contract EVM {
...
+    /// ValidationResult returns the result of COA ownership proof validation
+    access(all)
+    struct ValidationResult {
+        access(all)
+        let isValid: Bool
+        
+        access(all)
+        let problem: String?
+
+        init(_ isValid: Bool,_ problem: String? ) {
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:245]: Use argument labels, they make call-sites more readable:

```suggestion
        init(isValid: Bool, problem: String?) {
```

File: fvm/evm/stdlib/contract.cdc [hunk 2]
```diff
@@ -242,12 +257,15 @@ contract EVM {
...
-    ) {
+    ): ValidationResult {
 
         // make signature set first 
         // check number of signatures matches number of key indices
-        assert(keyIndices.length == signatures.length,
-               message: "key indices size doesn't match the signatures")
+        if keyIndices.length != signatures.length {
+            return ValidationResult(false,
+                "key indices size doesn't match the signatures"
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:266]:
```suggestion
            return ValidationResult(
                isValid: false,
                problem: "key indices size doesn't match the signatures"
```

File: fvm/evm/stdlib/contract.cdc [hunk 3]
```diff
@@ -278,18 +296,32 @@ contract EVM {
             signatureSet: signatureSet,
             signedData: signedData
         )
-        assert(isValid, message: "signatures not valid")
+
+        if !isValid{
+            return ValidationResult(false,
+                "the given signatures are not valid or provide enough weight"
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:302]:
```suggestion
            return ValidationResult(
                isValid: false,
                problem: "the given signatures are not valid or provide enough weight" 
```

File: fvm/evm/stdlib/contract.cdc [hunk 4]
```diff
@@ -278,18 +296,32 @@ contract EVM {
...
+        }
 
         let coaRef = acc.getCapability(path)
             .borrow<&EVM.CadenceOwnedAccount{EVM.Addressable}>()
-            ?? panic("could not borrow coa resource addressable capability")
+        
+        if coaRef == nil {
+             return ValidationResult(false,
+                    "could not borrow bridge account's resource"
+                )
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:312]:
```suggestion
             return ValidationResult(
                 isValid: false,
                 problem: "could not borrow bridge account's resource"
             )
```

File: fvm/evm/stdlib/contract.cdc [hunk 5]
```diff
@@ -278,18 +296,32 @@ contract EVM {
...
+        }
 
         // verify evm address matching
         var i = 0
-        for item in coaRef.address().bytes {
-            assert(item == evmAddress[i], message: "evm address mismatch")
+        var addr = coaRef!.address()
+        for item in addr.bytes {
+            if item != evmAddress[i] {
+                return ValidationResult(false, "evm address mismatch")
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:320]:
```suggestion
                return ValidationResult(
                    isValid: false,
                    problem: "EVM address mismatch"
                )
```

**Review by turbolent — APPROVED** [review]
👍

File: fvm/evm/stdlib/contract.cdc [hunk 6]
```diff
@@ -278,18 +296,32 @@ contract EVM {
...
+            }
             i = i +1
         }
-
+        
+        return ValidationResult(true, nil)
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:325]:
```suggestion
        return ValidationResult(
        	isValid: true,
        	problem: nil
        )
```

**Review by janezpodhostnik — APPROVED** [review]

**Pull request merged**